scalalib/src/scoverage/ScoverageModule.scala

```diff
@@ -0,0 +1,48 @@
+package mill.lib.scoverage
+
+import mill._
+import mill.scalalib._
+
+import _root_.scoverage.Serializer.{ coverageFile, deserialize }
+import _root_.scoverage.IOUtils.{ findMeasurementFiles, invoked }
+import _root_.scoverage.report.ScoverageHtmlWriter
+
+trait ScoverageModule { outer: ScalaModule =>
+
+  trait ScoverageCompile extends ScalaModule {
+    def dataDir: String
+    def sources = outer.sources
+    def resources = outer.resources
+    def scalaVersion = outer.scalaVersion()
+    def compileIvyDeps = outer.compileIvyDeps()
+    def ivyDeps = outer.ivyDeps() ++
+      Agg(ivy"org.scoverage::scalac-scoverage-runtime:1.3.1")
+    def scalacPluginIvyDeps = outer.scalacPluginIvyDeps() ++
+      Agg(ivy"org.scoverage::scalac-scoverage-plugin:1.3.1")
+    def scalacOptions = outer.scalacOptions() ++
+      Seq(s"-P:scoverage:dataDir:$dataDir")
+
+    def html() = T.command {
```

Comment:
My first thought was, that `html` is a bit too general wording. Maybe something like `report` is more self-explanatory?

Reply:
I think `htmlReport` as this leaves room to implement the other report formats that scoverage has like `xmlReport` for example